addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,294 @@
+	 * In the future, I'll consider only comparing "nearby" topic ids,
+	 * since one could scroll quite a few days back "on the same page".
+	 * Comparing posts made that many days apart, all using the most
+	 * recent topic id as a comparator could increase false positives.
+	 *
+	 * Inefficiency of a full highlightNecropostsIfOnSelectedForumPage
+	 * is probably not a concern here, browsing a forum page.
+	 */
+	async function onRowsChanged(mutationList, mutationObserver) {
+		await highlightNecropostsIfOnSelectedForumPage();
+	}
+	
+
+	/***********************************************************************
+	 * This is where the magic happens
+	 ***********************************************************************/
+	async function highlightNecropostsIfOnSelectedForumPage() {
+		let shouldRunOnThisPage = await isOnSelectedForumPage();
+		if (!shouldRunOnThisPage) {
+			return;
+		}
+		
+		let topics = await gatherTopics();
+		// console.log("Topic Count: " + topics.length);
+		let highestTopicOnThisPage = highestTopicIdFrom(topics);
+		// console.log("Highest TopicId: " + highestTopicOnThisPage);
+		let lowestNewTopicId = highestTopicOnThisPage - determineTopicInterval();
+		highlightTopicCellsWithTopicIdBelow(topics, lowestNewTopicId);
+	}
+	
+	/**
+	 * Returns an array of topic.
+	 * Each topic is:
+	 *    - topicId     int, used to compare topic "age" relative to other posts on the page
+	 *    - topicCell   Node that will be highlighted if too old
+	 *    - forum       String, usually the forum being listed. Comes from the row on a search page
+	 */
+	async function gatherTopics() {
+		let possibleTopicCells; // verified before forming a topic
+		if (isMobileSite) {
+			possibleTopicCells = document.querySelectorAll(".topic.list>li");
+		} else {
+			possibleTopicCells = document.querySelectorAll(".tcl");
+		}
+		// console.log("possibleTopicCells length: " + possibleTopicCells.length);
+		let topics = [];
+		let theForum = forumOrSearchPageName; // usually, but overridden per cell if on a search page
+		// console.log("theForum: " + theForum);
+		for (let i = 0; i < possibleTopicCells.length; i++) {
+			if (forumOrSearchPageName.includes(searchResultsPageName)) {
+				// the cell to the right of the topic cell lists the forum, in search results
+				theForum = possibleTopicCells[i].nextElementSibling.innerText;
+			}
+			let topic = {
+				topicId: await extractTopicIdFrom(possibleTopicCells[i]),
+				topicCell: possibleTopicCells[i],
+				forum: theForum
+			}
+			if (topic.topicId != 0) {
+				topics.push(topic);
+				// console.log("gatherTopics: " + topic.topicId + " in " + topic.forum);
+			}
+		}
+		return topics;
+	}
+	
+	/**
+	 * How many topics since the earliest on the same page to be considered a "necropost"
+	 */
+	function determineTopicInterval() {
+		if (!addon.settings.get("chooseCustomTopicInterval")) {
+			return defaultTopicInterval;
+		}
+		return addon.settings.get("customTopicInterval");
+	}
```

Comment:
`addon.settings.get` will **not** return a nullish object.

Reply:
`chooseCustomTopicInterval` is a Boolean setting.

Reply:
I think what you're saying is on 164 "!" can't be used to determine a "null" return.

That's not what I'm doing here. chooseCustomTopicInterval is a boolean setting that acts as an on/off switch for whether to use a default interval or a custom user value.

I'm using "!" as a _logical not_ to say "if the user **does not** enable the option to choose a custom interval", return the default. If you get past this guard, the user has switched "choose custom" to true, so use their custom interval.